### PR TITLE
New Addon: Remove Autocomplete from the Search Bar

### LINKED ISSUE #2983: Disable auto-complete on search bar
### Is your request related to a problem? Please describe.

No, it's not. It's a completely new useful addon that will... I think the title is pretty clear. An addon to disable the auto-complete on the navigation searchbar.

### Describe the solution you'd like

No.

### Describe alternatives you've considered

~~Either that or delete the search bar completely~~

### Additional context

No.

### PR DESCRIPTION
**Pull request by BroJac5246**
Resolves #2983

Note: I'm new to this. If I did anything wrong, please let me know and I'll do my best to fix it.
### Changes
This is a new addon that stops web browsers from autocompleting results into the search bar on the navigation bar.

### Reason for changes
One of the biggest things that has always bugged me while using Scratch is the search bar's autocomplete. While it's kind of small, it really agitates me to see the browser giving suggestions in that **horrid white popup**! I can't tell you the number of times I've looked at it and then immediately gone off to Chrome's settings to clear "Autofill form data" (I'm not kidding, I've done it multiple times)

Look at _that_:
Screenshot 2022-07-08 13 31 37
Ew!

That's why I created the Remove Search Bar Autocomplete addon. It sets `autocomplete` to `off`, so that we can all search in peace again.

It's a simple yet meaningful way to remove the clutter and improve the interface while searching on Scratch.

### Tests
Works on Chrome 103.
I haven't noticed any conflicts with other addons, including the `Scratch 2.0 → 3.0` addon.

### CODE REVIEW

**Comment by WorldLanguages** [comment]
Some browsers like to enable autocomplete even though it's set to autocomplete=off. We need to make sure this works in Chrome and Firefox in all situations.

@BroJac5246 Nice first PR! Expect follow up in a few days (this won't get into v1.27.0)

**Comment by mxmou** [comment]
A very similar PR was previously rejected: #3003

**Comment by BroJac5246** [comment]
> Some browsers like to enable autocomplete even though it's set to autocomplete=off. We need to make sure this works in Chrome and Firefox in all situations.

It appears to work fine for me in Chrome. I can't test it in Firefox, though (as I'm on a Chromebook).

**Comment by BroJac5246** [comment]
> A very similar PR was previously rejected: #3003

However, the idea itself clearly hasn't been rejected, as #2983 is still open, despite being directly linked in #3003.

Also, according to the official `Creating an Addon` Doc:
>Keep in mind we might request you to make some changes, however, we will probably accept your addon as long as it’s minimally suitable.

Unless the addon doesn't qualify as **minimally** suitable (and I don't see how it wouldn't), then the official docs directly imply that the addon should be accepted.

Besides, why shouldn't it be added? "Too small" (the reason used for #3003) is a horrible reason! Look at the paper clip. They're tiny, and cost less than 0.01 USD to make, but they've changed the way we organize things. Or the microSD card. They're the size of a fingernail, but can store over 2 TRILLION bytes. This addon takes up less than a KB, and could be utilized to remove an annoying kink in the search system.

**Comment by WorldLanguages** [comment]
> A very similar PR was previously rejected: #3003

The PR author closed the PR, not one of the org members. And the issue hasn't been closed. I still believe this is a pretty small addon, but I think it could be added as a non-featured addon.

I also remember this being suggested somewhere on feedback or the community discord server, but I can't find the messages.

**Comment by BroJac5246** [comment]
Also, can someone explain how to `Format code using Prettier / Format (pull_request)`? As a said, I'm new to this, and the error is annoying me. (Also, that detached head error may or may not be somewhat disturbing.)

File: addons/remove-search-bar-autocomplete/addon.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "Remove Autocomplete from Search Bar",
```
Comment (TheColaber) [inline addons/remove-search-bar-autocomplete/addon.json:2]: Use the casing we use for all addons. "Remove autocomplete from search bar"

File: addons/remove-search-bar-autocomplete/userscript.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default async function ({ addon, global, console }) {
```
Comment (TheColaber) [inline addons/remove-search-bar-autocomplete/userscript.js:1]: Enable prettier at https://github.com/BroJac5246/ScratchAddons/actions

Comment (BroJac5246) [reply addons/remove-search-bar-autocomplete/userscript.js:1]: I _think_ I got it? Can you check real quick? <sub>Sorry, like I said, I'm new to this.</sup>

Comment (RIMOPA) [reply addons/remove-search-bar-autocomplete/userscript.js:1]: > I _think_ I got it? Can you check real quick? Sorry, like I said, I'm new to this.

Yes you got it

File: addons/remove-search-bar-autocomplete/addon.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    {
+      "name": "jbthepig",
+      "link": "https://scratch.mit.edu/users/jbthepig"
+    }
+  ],
+  "dynamicEnable": true,
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["https://scratch.mit.edu/*"]
```
Comment (TheColaber) [inline addons/remove-search-bar-autocomplete/addon.json:15]: I recommend adding `"runAtComplete": false` underneath `matches` just so it starts running a tad bit faster.

File: addons/remove-search-bar-autocomplete/addon.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "Stops web browsers from autocompleting suggestions in the navigation bar's search bar.",
+  "tags": ["community"],
+  "credits": [
+    {
+      "name": "jbthepig",
+      "link": "https://scratch.mit.edu/users/jbthepig"
+    }
+  ],
+  "dynamicEnable": true,
+  "userscripts": [
```
Comment (TheColaber) [inline addons/remove-search-bar-autocomplete/addon.json:12]: Could you attempt adding dynamicDisable to this addon? You can use an [`addon.self.addEventListener` on "disabled" and "reenabled"](https://scratchaddons.com/docs/reference/addon-api/addon.self/#disabled) to achieve this. If not, I can take care of it for you :)

File: addons/remove-search-bar-autocomplete/addon.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    {
+      "url": "userscript.js",
+      "matches": ["https://scratch.mit.edu/*"]
+    }
+  ],
+  "versionAdded": "1.27.0"
```
Comment (TheColaber) [inline addons/remove-search-bar-autocomplete/addon.json:18]: Just a reminder to change this to 1.28.0, at least before this gets merged.

**Review by TheColaber — CHANGES_REQUESTED** [review]
Well done on your first addon 👍 I also did test this and it worked fine. I'll test it again once dynamically disabling and re-enabling are added.

**Comment by BroJac5246** [comment]
I just pushed some new commits, and would like them reviewed for some extra certainty that they work as intended :)

**Review by TheColaber — APPROVED** [review]

**Comment by DNin01** [comment]
Please note that you'll need to request a merge from your own branch instead of the "master" branch. You can make a new one on the branches list of your fork. But after that, your pull request should be ready.

[Here's some help](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch) if you need it.

**Comment by BroJac5246** [comment]
> Please note that you'll need to request a merge from your own branch instead of the "master" branch. You can make a new one on the branches list of your fork. But after that, your pull request should be ready.
> 
> [Here's some help](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch) if you need it.

Can you explain? I have a new branch at https://github.com/BroJac5246/ScratchAddons/tree/search-bar-autocomplete, but what do I need to do next? Sorry, I'm new to this.

**Comment by DNin01** [comment]
> Can you explain? I have a new branch at https://github.com/BroJac5246/ScratchAddons/tree/search-bar-autocomplete, but what do I need to do next? Sorry, I'm new to this.

Oh, yeah. Next, you can create another pull request, but this time requesting to merge changes from your new branch. It should show "`base repository: ScratchAddons/ScratchAddons` `base: master` ← `head repository: BroJac5246/ScratchAddons` `compare: search-bar-autocomplete`", I believe. Afterwards, you can close this pull request.

**Comment by BroJac5246** [comment]
Closing so that we can merge from #4819

**Comment by Weredime** [comment]
> Please note that you'll need to request a merge from your own branch instead of the "master" branch. You can make a new one on the branches list of your fork. But after that, your pull request should be ready.
> 
> [Here's some help](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch) if you need it.

??????
You can still do this, W_L doesn't care too much. But it's still good practice to make a new branch though.

@BroJac5246 After #4819, delete your fork and fork this repository again. That ensures master is clean.

**Comment by BroJac5246** [comment]
> > Please note that you'll need to request a merge from your own branch instead of the "master" branch. You can make a new one on the branches list of your fork. But after that, your pull request should be ready.
> > [Here's some help](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch) if you need it.
> 
> ?????? You can still do this, W_L doesn't care too much. But it's still good practice to make a new branch though.
> 
> @BroJac5246 After #4819, delete your fork and fork this repository again. That ensures master is clean.

So, if I'm understanding correctly, it's a best practice to merge from your own branch, but it's not required? And after the merge, I should create a new fork? I just want to make sure :)

**Comment by DNin01** [comment]
> > > Please note that you'll need to request a merge from your own branch instead of the "master" branch. You can make a new one on the branches list of your fork. But after that, your pull request should be ready.
> 
> > > [Here's some help](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch) if you need it.
> 
> > 
> 
> > ?????? You can still do this, W_L doesn't care too much. But it's still good practice to make a new branch though.
> 
> > 
> 
> > @BroJac5246 After #4819, delete your fork and fork this repository again. That ensures master is clean.
> 
> 
> 
> So, if I'm understanding correctly, it's a best practice to merge from your own branch, but it's not required? And after the merge, I should create a new fork? I just want to make sure :)

I'm not exactly sure why it's obligated. But it does allow you to more easily manage your modifications: keeping track of and working with 4 branches makes things easier than having 4 forks, IMO.

And yes, after the merge, I would create a fresh new fork to ensure the `master` branch is unmodified.

**Pull request closed without merging**